src/middlewared/middlewared/plugins/etc.py

```diff
@@ -510,6 +483,16 @@ async def generate_checkpoint(self, checkpoint):
     async def get_checkpoints(self):
         return self.checkpoints
 
+    async def get_user_id(self, username):
+        if not self.SYS_USERS:
```

Comment:
I have a few proposals:
1. Move these methods to `user`/`group` plugins since they can be used by other plugins (e.g. webdav)
2. Name methods like `get_internal_user_id` to avoid confusion (someone might try to use them to retrieve normal user ids)
3. Warm up this cache on `user`/`group` plugin initialization because it's gonna be used anyway

Reply:
`get_service_uid` / `get_service_gid` maybe?